### PR TITLE
feat: custom bootstrap code

### PR DESCRIPTION
**Pull request by ncitron**
addresses #195

### CODE REVIEW

File: huff_codegen/src/lib.rs [hunk 1]
```diff
@@ -525,8 +525,15 @@ impl Codegen {
             )
         };
 
+        let has_custom_bootstrap = hex::decode(constructor_bytecode).unwrap().contains(&0xf3);
```
Comment (clabby) [inline huff_codegen/src/lib.rs:528]: :+1:

File: huff_codegen/src/lib.rs [hunk 2]
```diff
@@ -525,8 +525,15 @@ impl Codegen {
...
+
+        let bootstrap_code = if has_custom_bootstrap {
+            String::new()
```
Comment (clabby) [inline huff_codegen/src/lib.rs:531]:
```suggestion
            String::default()
```

File: huff_codegen/tests/churn.rs [hunk 3]
```diff
@@ -28,6 +28,31 @@ fn churns_into_bytecode() {
...
+    let churn_res =
+        cg.churn(Arc::new(FileSource::default()), inputs, main_bytecode, constructor_bytecode);
+
+    assert!(churn_res.is_ok());
+    assert_eq!(
+        churn_res.unwrap().bytecode,
+        "64600160020160005260056000f36001600216".to_lowercase()
+    );
+}
+
```
Comment (clabby) [inline huff_codegen/tests/churn.rs:55]: Nice!

**Review by clabby — COMMENTED** [review]
Great feature! Wondering if we should flesh it out a bit more so that if the constructor has a custom bootstrap, the `MAIN` macro isn't required. Developers can always just make an empty one though, not sure if its worth it. Wdyt @abigger87 @JetJadeja?

**Review by abigger87 — APPROVED** [review]
LGTM

**Pull request merged**